quarto/Group_10_Analysis.qmd

```diff
@@ -0,0 +1,214 @@
+#| echo: true
+#| warning: false
+
+# Creating a new column to categorize ratings higher then 7 or not
+library(tidyverse)
+
+df <- df %>% 
+  mutate(rating_new = ifelse(rating <= 7, "<= 7", "> 7"))
+```
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Filling missing values with mode
+
+modes <- function(x) {
+  ux <- unique(x)
+  ux[which.max(tabulate(match(x, ux)))]
+}
+
+columns <- c('length')
+
+for (column in columns) {
+  # get the mode of each column
+  mode_value <- as.character(modes(df[[column]]))
+  # replace missing values with modes
+  df[[column]][is.na(df[[column]])] <- mode_value
+}
+
+# show the result of filling the missing values
+sapply(df[columns], function(x) sum(is.na(x)))
+```
+
+# Exploratory Data Analysis
+
+Draw scatter plots of scores and variables for preliminary exploratory analysis
+
+```{r}
+#| echo: true
+#| warning: false
+
+library(ggplot2)
+
+summary(df)
+
+ggplot(df, aes(x = year, y = rating)) +
+  geom_point() +
+  theme_minimal() +
+  labs(title = "Rating vs Year", x = "Year", y = "Rating")
+
+ggplot(df, aes(x = length, y = rating)) +
+  geom_point() +
+  theme_minimal() +
+  labs(title = "Rating vs Length", x = "Length", y = "Rating")
+
+ggplot(df, aes(x = budget, y = rating)) +
+  geom_point() +
+  theme_minimal() +
+  labs(title = "Rating vs Budget", x = "Budget", y = "Rating")
+
+ggplot(df, aes(x = votes, y = rating)) +
+  geom_point() +
+  theme_minimal() +
+  labs(title = "Rating vs Votes", x = "Votes", y = "Rating")
+
+```
+
+```{r}
+# Calculate the average rating for each type of movie
+library(dplyr)
+average_ratings <- df %>%
+  group_by(genre) %>%
+  summarise(average_rating = mean(rating, na.rm = TRUE)) %>%
+  ungroup() %>%  # Make sure that grouping is removed so that all results can be sorted
+  arrange(desc(average_rating))
+
+# show result
+print(average_ratings)
+
+```
+
+```{r}
+# Calculate the correlation coefficient between all the variables in df
+# Ensure that the data only contains numeric columns
+
+numeric_df <- df[sapply(df, is.numeric)]
+correlation_matrix <- cor(numeric_df, use = "complete.obs")# The 'use' parameter handles missing values
+
+print(correlation_matrix)
+
+```
+
+```{r}
+#| echo: true
+#| warning: false
+
+library(ggplot2)
+library(gridExtra)
+
+p1 <- ggplot(data = df, mapping = aes(x = rating_new, y = year)) +
+  geom_boxplot(fill = "steelblue")
+
+p2 <- ggplot(data = df, mapping = aes(x = rating_new, y = length)) +
+  geom_boxplot(fill = "steelblue")
+
+p3 <- ggplot(data = df, mapping = aes(x = rating_new, y = budget)) +
+  geom_boxplot(fill = "steelblue")
+
+p4 <- ggplot(data = df, mapping = aes(x = rating_new, y = votes)) +
+  geom_boxplot(fill = "steelblue")
+
+p5 <- ggplot(df, aes(x= genre,  y = ..prop.., group=rating_new, fill=rating_new)) + 
+  geom_bar(position="dodge", stat="count") +
+  labs(y = "Proportion")
+
+grid.arrange(p1, p2, p3, p4, ncol=2)#This part of the code has reported an error, but I can not solve it, I do not know what the problem is
```

Comment:
This part of the code has reported an error, but I can not solve it, I do not know what the problem is